### PR TITLE
Add product utility unit tests

### PR DESCRIPTION
**Pull request by deeglaze**
Improves test coverage, but also expands GetAttestationFromReport to fill in the Product field of the Attestation message from the fetched VCEK/VLEK certificate's extension.

### CODE REVIEW

File: verify/verify.go [hunk 1]
```diff
@@ -772,6 +770,23 @@ func GetAttestationFromReport(report *spb.Report, options *Options) (*spb.Attest
 	if err := fillInAttestation(result, options); err != nil {
 		return nil, err
 	}
+	// Attempt to fill in the product field of the attestation. Don't error at this
+	// point since this is not validation.
+	info, _ := abi.ParseSignerInfo(report.SignerInfo)
+	var exts *kds.Extensions
+	parse := func(der []byte) *x509.Certificate {
+		out, _ := x509.ParseCertificate(der)
```
Comment (kongoshuu) [inline verify/verify.go:778]: nit: It makes sense not to error here. But I wonder if there's an error from ParseCertificate (or kds.ParseProductName), shall we add a log for them? Or are we relying on the real error message to include those info?

Comment (deeglaze) [reply verify/verify.go:778]: The real error message will happen during verify.SnpAttestation if a certificate doesn't parse.

Comment (deeglaze) [reply verify/verify.go:778]: Is there a change you're requesting?

Comment (kongoshuu) [reply verify/verify.go:778]: SGTM.

**Pull request merged**